amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -135,25 +152,35 @@ private void startProducer() {
         if (consumerConfig.getReshardFactorList() != null) {
             log.info("----Reshard Config found: {}", consumerConfig.getReshardFactorList());
 
-            final StreamScaler s = new StreamScaler(
-                    kinesisClient,
-                    consumerConfig.getStreamName(),
-                    consumerConfig.getReshardFactorList(),
-                    consumerConfig
-            );
+            for (String streamName : consumerConfig.getStreamNames()) {
+                final StreamScaler streamScaler = new StreamScaler(kinesisClientForStreamOwner, streamName,
+                        consumerConfig.getReshardFactorList(), consumerConfig);
 
-            // Schedule the stream scales 4 minutes apart with 2 minute starting delay
-            for (int i = 0; i < consumerConfig.getReshardFactorList().size(); i++) {
-                producerExecutor.schedule(s, (4 * i) + 2, TimeUnit.MINUTES);
+                // Schedule the stream scales 4 minutes apart with 2 minute starting delay
+                for (int i = 0; i < consumerConfig.getReshardFactorList()
+                                                  .size(); i++) {
+                    producerExecutor.schedule(streamScaler, (4 * i) + 2, TimeUnit.MINUTES);
+                }
             }
         }
     }
 
-    private void setUpConsumerResources() throws Exception {
+    private void setUpConsumerResources(Map<Arn, Arn> streamToConsumerArnsMap) throws Exception {
         // Setup configuration of KCL (including DynamoDB and CloudWatch)
-        final ConfigsBuilder configsBuilder = consumerConfig.getConfigsBuilder();
+        final ConfigsBuilder configsBuilder = consumerConfig.getConfigsBuilder(streamToConsumerArnsMap);
+
+        // For polling mode in both CAA and non CAA, set retrievalSpecificConfig to use PollingConfig
+        // For SingleStreamMode EFO CAA, must set the retrieval config to specify the consumerArn in FanoutConfig
+        // For MultiStream EFO CAA, the consumerArn can be set in StreamConfig
+        if (consumerConfig.getRetrievalMode().equals(RetrievalMode.POLLING)) {
+            retrievalConfig = consumerConfig.getRetrievalConfig(configsBuilder, null);
+        } else if (consumerConfig.isCrossAccount()) {
+            retrievalConfig = consumerConfig.getRetrievalConfig(configsBuilder,
+                    streamToConsumerArnsMap);
+        } else {
+            retrievalConfig = configsBuilder.retrievalConfig();
+        }
```

Comment:
should we simplify this to:
```
retrievalConfig = consumerConfig.getRetrievalConfig(configsBuilder, streamToConsumerArnsMap);
```
looks like the `KCLAppConfig#getRetrievalConfig` method already contains most of the necessary logic, might just need to move the `isCrossAccount` check to there to achieve the same behavior?